Sources/ApolloCodegenLib/ApolloCodegen.swift

```diff
@@ -104,52 +70,101 @@ public class ApolloCodegen {
     return try frontend.compile(schema: graphqlSchema, document: mergedDocument)
   }
 
-  static func fileGenerators(
-    for objectTypes: OrderedSet<GraphQLObjectType>,
-    directoryPath path: String
-  ) -> [TypeFileGenerator] {
-    return objectTypes.map({ graphqlObjectType in
-      TypeFileGenerator(objectType: graphqlObjectType, directoryPath: path)
-    })
-  }
+  /// Generates Swift files for the compiled schema, ir and configured output structure.
+  static func generateFiles(
+    compilationResult: CompilationResult,
+    ir: IR,
+    config: ApolloCodegenConfiguration,
+    fileManager: FileManager = FileManager.default
```

Comment:
I'm not a huge fan of having to expose `fileManager` in many more function definitions but I don't see another way around it for mocked testing; the alternative is actually writing to the file system.

At least this function is _internal_ so users calling ApolloCodegen won't ever know this exists.

Reply:
Yeah, it's annoying but necessary. We should not be writing to the file system in unit tests.